app/controllers/users_controller.rb

```diff
@@ -67,17 +69,17 @@ def user_params
     end
   end
 
-  def require_admin
-    unless current_user.admin?
-      set_flash_message :alert, :must_be_admin
-      redirect_to root_path
-    end
-  end
-
   def remove_unused_password_pair_from_params
     if params[:user][:password].blank?
       params[:user].delete(:password)
       params[:user].delete(:password_confirmation)
     end
   end
+
+  def require_admin
```

Comment:
The edit logic is a bit tricky.  A user can edit his own entry and an admin can edit anyone's entry.  If I add :edit to the before filter, then the user cannot edit his own entry.